### ISSUE TITLE
[0xProject/connect] WebSocketOrderbookChannel can only be constructed when running in node.js environment.

### ISSUE THREAD

**Issue by fbielejec** [0xProject/0x-monorepo#477]
The constructor [here](https://github.com/0xProject/0x-monorepo/blob/development/packages/connect/src/ws_orderbook_channel.ts#L48) calls the node only [implementation](https://github.com/theturtle32/WebSocket-Node#client-example).

This could easily be made to work in the browser by using the [W3C implementation](https://github.com/theturtle32/WebSocket-Node#client-example-using-the-w3c-websocket-api) instead:

```this._client = new WebSocket.w3cwebsocket();```

or even using Browserify / Webpack with a process.browser field and a prepublish step:

```
if (process.env.BROWSER) {
  this._client = new WebSocket.w3cwebsocket();
} else {
  this._client = new WebSocket.client();
}
```

**Comment by krishishah** [comment]
Any updates on this?

**Comment by BMillman19** [comment]
@fbielejec @krishishah i looked into it a bit, its not a one line fix as suggested above and will require a bit of a refactor to accommodate,  I will try to get to it this week. But as always feel free to submit a PR to solve it as well

**Comment by BMillman19** [comment]
This should be fixed as of 0x connect version 1.0.0 which should be published by the end of this week slash early next week, check out this issue: https://github.com/0xProject/0x-monorepo/issues/558 for a sample of how to use the new API

**Comment by rajqumar** [comment]
Hi @BMillman19 ,
I tried implementing the above code, everything works except I get error for this code
```
    const relayerWsApiUrl = 'http://localhost:3001/v0';
    const handler: OrderbookChannelHandler = CustomOrderbookChannelHandler;
    const orderbookChannel: OrderbookChannel = await orderbookChannelFactory.createWebSocketOrderbookChannelAsync(relayerWsApiUrl, handler);
```


I solved this Error: 
Type 'typeof CustomOrderbookChannelHandler' is not assignable to type 'OrderbookChannelHandler'.
  Property 'onSnapshot' is missing in type 'typeof CustomOrderbookChannelHandler'.

But stuck at this: 
node_modules/@0xproject/connect/node_modules/@0xproject/utils/lib/fetchAsync.d.ts(2,63): error TS2304: Cannot find name 'RequestInit'.
node_modules/@0xproject/connect/node_modules/@0xproject/utils/lib/fetchAsync.d.ts(2,107): error TS2304: Cannot find name 'Response'.
